lib/yam/agama/agama_base.pm

```diff
@@ -0,0 +1,20 @@
+## Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# Summary: base class for Agama tests
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+package yam::agama::agama_base;
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use y2_base 'save_upload_y2logs';
+
+sub post_fail_hook {
+    $testapi::password = 'linux';
+    select_console 'root-console';
+    y2_base::save_upload_y2logs;
```

Comment:
Could you also collect journalctl log in post_fail_hook?
script_run('journalctl -u agama-auto');

Another question is, when system is installed correctly with password nots3** , then this post_fail_hook will failed. How to handle this kind of situation?

Reply:
BTW my PR link is:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/16999

Reply:
> Could you also collect journalctl log in post_fail_hook? script_run('journalctl -u agama-auto');
> 
> Another question is, when system is installed correctly with password nots3** , then this post_fail_hook will failed. How to handle this kind of situation? such as this case https://openqa.opensuse.org/tests/3290466#step/auto/18

save_upload_y2logs already uploads all the output from journalctl, so it should include as well include agama-auto unit's logs, I assume. For the second question, it's a good point, we probably need to set this in the module, I'm checking if that works.

Reply:
So the answer to that is now, after install we will switch to a different module with a different base. Agama is the installer, so it should only be used in the installer.